java/code/src/com/suse/manager/hub/test/HubManagerTest.java

```diff
@@ -626,4 +641,82 @@ else if (char.class.equals(clazz)) {
 
         return null;
     }
+
+    @Test
+    public void checkApiManagerinfo() throws Exception {
+        String apiUnderTest = "/hub/managerinfo";
+
+        String answerOK = (String) simulateHubControllerApi(apiUnderTest, HttpMethod.get, IssRole.HUB);
+        ManagerInfoJson mgrInfo = Json.GSON.fromJson(answerOK, ManagerInfoJson.class);
+        assertEquals("5.1.0", mgrInfo.getVersion());
+    }
+
+    @Test
+    public void ensureApiManagerinfoNotWorkingOnPut() {
+        String apiUnderTest = "/hub/managerinfo";
+
+        assertThrows(IllegalStateException.class,
+                () -> simulateHubControllerApi(apiUnderTest, HttpMethod.put, IssRole.HUB));
+    }
+
+    @Test
+    public void ensureApiManagerinfoNotWorkingOnPeripheral() throws Exception {
+        String apiUnderTest = "/hub/managerinfo";
+
+        String answerKO = (String) simulateHubControllerApi(apiUnderTest, HttpMethod.get, IssRole.PERIPHERAL);
+        ResultJson<?> resultKO = Json.GSON.fromJson(answerKO, ResultJson.class);
+        assertFalse(resultKO.isSuccess());
+        assertEquals("Token does not allow access to this resource", resultKO.getMessages().get(0));
+    }
+
+    private Object simulateHubControllerApi(String apiEndpoint, HttpMethod httpMethod, IssRole role)
+            throws Exception {
+        String dummyServerFqdn = "dummy-server.unit-test.local";
+
+        HubController syncController = new HubController();
+        syncController.initRoutes();
+
+        Token dummyServerToken = new IssTokenBuilder(dummyServerFqdn)
+                .usingServerSecret()
+                .build();
+        String authBearerToken = dummyServerToken.getSerializedForm();
+
+        hubFactory.saveToken(dummyServerFqdn, authBearerToken, TokenType.ISSUED, dummyServerToken.getExpirationTime());
+        switch (role) {
+            case HUB:
+                hubFactory.save(new IssHub(dummyServerFqdn, ""));
+                break;
+            case PERIPHERAL:
+                hubFactory.save(new IssPeripheral(dummyServerFqdn, ""));
+                break;
+            default:
+                throw new IllegalArgumentException("unsupported role " + role.getLabel());
+        }
+
+        return simulateApiEndpointCall(apiEndpoint, httpMethod, authBearerToken);
+    }
+
+    //maybe this method could be moved to SparkTestUtils?
+    private Object simulateApiEndpointCall(String apiEndpoint, HttpMethod httpMethod, String authBearerToken)
+            throws Exception {
+        Optional<RouteMatch> routeMatch = spark.Spark.routes()
```

Comment:
would be a option to use the route find method, which would be able to handle wildcard?
In the URL you could use the expected URL to be used (like: `/systems/1000001/bla`) which would math the route pattern like `/systems/:ID/bla`.
You can see an example in here: https://github.com/rjmateus/uyuni/blob/uyuni_hack_authorization_check/java/code/src/com/redhat/rhn/frontend/servlets/AccessFilter.java#L116-L118